src/rotator_library/providers/qwen_auth_base.py

```
@@ -380,6 +395,24 @@ async def _refresh_token(self, path: str, force: bool = False) -> Dict[str, Any]
                             timeout=30.0,
                         )
                         response.raise_for_status()
+
+                        # [WAF DETECTION] Alibaba Cloud WAF may return HTTP 200
+                        # with HTML content instead of JSON. Detect and retry.
+                        content_type = response.headers.get("content-type", "")
+                        if "json" not in content_type.lower():
+                            lib_logger.warning(
+                                f"Token refresh for '{Path(path).name}' returned non-JSON "
+                                f"(content-type: {content_type}), likely WAF block. "
+                                f"Attempt {attempt + 1}/{max_retries}."
```

Ruff (0.15.9)

[warning] Logging statement uses f-string

(G004)